lib/Network/OAuth2/Server/Types/Error.hs

```diff
@@ -218,3 +225,15 @@ instance FromJSON ErrorDescription where
             Just s -> return s
 
 --------------------------------------------------------------------------------
+
+-- Convenience functions for throwing errors
+
+do TyConI (DataD _ _ _ cs _) <- reify ''ErrorCode
```

Comment:
This PR is overall a vast improvement. This bit, however, I cannot make sense of at a glance.

Could you document what it outputs? I can guess, but one example ErrorCode would be nice.